src/man/sssd-secrets.5.xml

```diff
@@ -0,0 +1,446 @@
+                <listitem>
+                <para>
+                    If set, the secrets responder would put a header with this name
+                    into the HTTP request with the value defined in the
+                    <quote>auth_header_value</quote> configuration option.
+                </para>
+                <para>
+                    Example: MYSECRETNAME
+                </para>
+                </listitem>
+            </varlistentry>
+            <varlistentry>
+                <term>auth_header_value (string)</term>
+                <listitem>
+                <para>
+                    The value sssd-secrets would use for the
+                    <quote>auth_header_name</quote>.
+                </para>
+                <para>
+                    Example: mysecret
+                </para>
+                </listitem>
+            </varlistentry>
+            <varlistentry>
+                <term>forward_headers (list of strings)</term>
+                <listitem>
+                <para>
+                    The list of HTTP headers to forward to the Custodia server
+                    together with the request.
+                </para>
+                <para>
+                    Default: not set
+                </para>
+                </listitem>
+            </varlistentry>
+        </variablelist>
+    </refsect1>
+    <refsect1 id='restapi'>
+        <title>USING THE REST API</title>
+        <para>
+            This section lists the available commands and includes examples using the
+            <citerefentry>
+                <refentrytitle>curl</refentrytitle>
+                <manvolnum>1</manvolnum>
+            </citerefentry> utility.
+            All requests towards the proxy provider must set the Content
+            Type header to <quote>application/json</quote>. In addition,
+            the local provider also supports Content Type set to
+            <quote>application/octet-stream</quote>.
+            Secrets stored with requests that set the Content Type header
+            to <quote>application/octet-stream</quote> are base64-encoded
+            when stored and decoded when retrieved, so it's not possible to
+            store a secret with one Content Type and retrieve with another.
+            The secret URI must begin with <filename>/secrets/</filename>.
+        </para>
+            <variablelist>
+                <varlistentry>
+                    <term>Listing secrets</term>
+                    <listitem>
+                    <para>
+                        To list the available secrets, send a HTTP GET request
+                        with a trailing slash appended to the container path.
+                    </para>
+                    <para>
+                        Example:
+                        <programlisting>
+curl -H "Content-Type: application/json" \
+     --unix-socket /var/run/secrets.socket \
+     -XGET http://localhost/secrets/
```

Comment:
IIRC curl had support for HTTP request via unix sockets even in older versions
But cURL 7.50+ requires a valid URL to be provided, including a hostname.
older versions could use `-XGET http:/secrets/`.

we needn't mention that in man page but we can mention requirement  for "cURL 7.50+"